applications/GeoMechanicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_mass_and_damping.h

```diff
@@ -0,0 +1,658 @@
+ * @ingroup KratosCore
+ * @brief Current class provides an implementation for builder and solving operations, while the global
+ * mass and damping matrices are stored.
+ * @details When the LHS is build, the global mass and damping matrices are build seperately. When building the RHS,
+ * the mass and damping matrices are multiplied with respectively the second and first derivative vector to calculate
+ * the mass and damping contribution.
+ * The RHS is constituted by the unbalanced loads (residual)
+ * Degrees of freedom are reordered putting the restrained degrees of freedom at
+ * the end of the system ordered in reverse order with respect to the DofSet.
+ * Imposition of the dirichlet conditions is naturally dealt with as the residual already contains
+ * this information.
+ * Calculation of the reactions involves a cost very similar to the calculation of the total residual.
+ * This class is intended to be used when the mass and damping matrices are constant throughout the iterations,
+ * using this class, when rebuilding the LHS every iteration is not efficient.
+ * @tparam TSparseSpace The sparse system considered
+ * @tparam TDenseSpace The dense system considered
+ * @tparam TLinearSolver The linear solver considered
+ * @author Aron Noordam
+ */
+template<class TSparseSpace,
+         class TDenseSpace, //= DenseSpace<double>,
+         class TLinearSolver //= LinearSolver<TSparseSpace,TDenseSpace>
+         >
+class ResidualBasedBlockBuilderAndSolverWithMassAndDamping
+    : public ResidualBasedBlockBuilderAndSolver< TSparseSpace, TDenseSpace, TLinearSolver >
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Definition of the flags
+    KRATOS_DEFINE_LOCAL_FLAG( SILENT_WARNINGS );
+
+    /// Definition of the pointer
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBlockBuilderAndSolverWithMassAndDamping);
+
+    /// Definition of the base class
+    using BaseType = ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>;
+
+    /// Definition of the classes from the base class
+    using TSchemeType = typename BaseType::TSchemeType;
+    using TSystemMatrixType = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+    using NodesArrayType = typename BaseType::NodesArrayType;
+    using ElementsArrayType = typename BaseType::ElementsArrayType;
+    using ConditionsArrayType = typename BaseType::ConditionsArrayType;
+
+    /// Additional definitions
+    using ElementsContainerType = PointerVectorSet < Element, IndexedObject>;
+    using EquationIdVectorType = Element::EquationIdVectorType;
+
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     */
+    explicit ResidualBasedBlockBuilderAndSolverWithMassAndDamping() : BaseType()
+    {
+    }
```

Comment:
To be in line with [C.80: Use =default if you have to be explicit about using the default semantics](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c80-use-default-if-you-have-to-be-explicit-about-using-the-default-semantics) from the C++ Core Guidelines I would change this as follows:
```suggestion
    ResidualBasedBlockBuilderAndSolverWithMassAndDamping() = default;
```